cai_causal_graph/causal_graph.py

```diff
@@ -534,6 +542,75 @@ def is_empty(self) -> bool:
         """Return True if there are no nodes and edges. False otherwise."""
         return len(self.nodes) == 0 and len(self.edges) == 0
 
+    def _check_node(self, identifier: NodeLike) -> str:
+        identifier = Node.identifier_from(identifier)
+
+        if identifier in self._nodes_by_identifier:
+            raise CausalGraphErrors.NodeDuplicatedError(f'Node already exists: {identifier}')
+        return identifier
+
+    def _prepare_nodes(self, source: NodeLike, destination: NodeLike) -> Tuple[List[Node], List[Node]]:
+        """Prepare the source and destination nodes for adding an edge."""
+        source_meta = None
+        if isinstance(source, HasMetadata):
+            source_meta = source.get_metadata()
+
+        destination_meta = None
+        if isinstance(destination, HasMetadata):
+            destination_meta = destination.get_metadata()
```

Comment:
How about `destination_meta = destination.get_metadata() if isinstance(destination, HasMetaData) else None`. Should easily fit into 120 characters.